### PR TITLE
v2.0 Add unit tests for  `as_flex_table`

### PR DESCRIPTION
**Pull request by edelarua**
**What changes are proposed in this pull request?**

Add v2.0 unit tests for `as_flex_table`.

Part of https://github.com/insightsengineering/sme-tasks/issues/689

Closes https://github.com/insightsengineering/sme-tasks/issues/695

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed by running `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: tests/testthat/test-as_flex_table.R [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  expect_silent(ft_tbl_uvregression <- my_tbl_uvregression |> as_flex_table())
+
+  expect_equal(
+    my_tbl_uvregression |> as_tibble(col_labels = FALSE),
+    ft_tbl_uvregression$body$dataset |> as_tibble(),
+    ignore_attr = c("class", "names")
+  )
+
+  # spanning_tbl
+  expect_equal(
```
Comment (ddsjoberg) [inline tests/testthat/test-as_flex_table.R:114]: What are we testing with this one? Once we convert to a tibble, the spanning headers are gone?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
This is amazing! 

Do we have tests for the bold/italic using markdown syntax in the header/spanning header? If not, can we add something for that, please?
image


🥇

File: tests/testthat/test-as_flex_table.R (hunk 1 above)
Comment (edelarua) [reply tests/testthat/test-as_flex_table.R]: You're correct - I have removed this.

**Comment by edelarua** [comment]
@ddsjoberg I have added a test for markdown syntax in headers. This is ready for another review whenever!

**Review by ddsjoberg — APPROVED** [review]
Stunning! 🤩🤩🤩 Thank you !!!!!

**Pull request merged**